src/plots/cartesian/axes.js

```diff
@@ -2326,54 +2263,59 @@ axes.doTicks = function(gd, axid, skipTitle) {
         }
         return drawLabels(ax._axislayer, ax._pos);
     }
-    else {
+    else if(fullLayout._has('cartesian')) {
         subplots = axes.getSubplots(gd, ax);
-        var alldone = subplots.map(function(subplot) {
+
+        // keep track of which subplots (by main conteraxis) we've already
+        // drawn grids for, so we don't overdraw overlaying subplots
+        var finishedGrids = {};
```

Comment:
note that you can't do something simple like pick the main axis for each overlaying subplot, as that subplot may not exist (say for example you have `x2y2` overlaying `xy` - for `y2` the main counteraxis is `x` but there is no `xy2` subplot. So instead I just make sure we only draw the grid once for any given main axis.